app/angular/src/client/preview/angular-beta/ComputesTemplateFromComponent.ts

```diff
@@ -127,5 +128,55 @@ export const computesTemplateSourceFromComponent = (
       ? ` ${initialOutputs.map((i) => `(${i})="${i}($event)"`).join(' ')}`
       : '';
 
-  return `<${ngComponentMetadata.selector}${templateInputs}${templateOutputs}></${ngComponentMetadata.selector}>`;
+  const template = buildTemplate(ngComponentMetadata.selector);
+  return `<${template.openTag}${templateInputs}${templateOutputs}></${template.closeTag}>`;
+};
+
+const buildTemplate = (
+  selector: string
+): {
+  openTag?: string;
+  closeTag?: string;
+} => {
+  const templates = [
+    {
+      // Match element selectors with optional chained attributes or classes
+      re: /^([\w\d-_]+)(?:(?:\[([\w\d-_]+)(?:=(.+))?\])|\.([\w\d-_]+))?/,
+      openTag: (matched: string[]) => {
+        let template = matched[1];
+        if (matched[2]) {
+          template += ` ${matched[2]}`;
+        }
+        if (matched[3]) {
+          template += `="${matched[3]}"`;
+        }
+        if (matched[4]) {
+          template += ` class="${matched[4]}"`;
+        }
+        return template;
+      },
+      closeTag: (matched: string[]) => `${matched[1]}`,
+    },
+    {
+      re: /^\.(.+)/,
+      openTag: (matched: string[]) => `div class="${matched[1]}"`,
+      closeTag: (matched: string[]) => `div`,
+    },
+    {
+      re: /^\[([\w\d-_]+)(?:=(.+))?\]/,
+      openTag: (matched: string[]) => `div ${matched[1]} ${matched[2] ? `="${matched[2]}"` : ''}`,
+      closeTag: (matched: string[]) => `div`,
+    },
+  ];
+
+  return templates.reduce((acc, template) => {
+    const matched = selector.match(template.re);
+    if (matched) {
+      return {
+        openTag: template.openTag(matched).trim(),
+        closeTag: template.closeTag(matched),
+      };
+    }
+    return acc;
+  }, {});
```

Comment:
what do you think about starting with a default value? 
like `storybook-default-selector` ? (or something like that)
In case of an error it could be a workaround? 

or we could display a clearer error if the open/close tag is not found ?

Reply:
N/A with new commit using the regex you mentioned which should always return a selector.